kodilla-good-patterns/src/main/java/com/kodilla/good/patterns/flight/FlightFinder.java

```diff
@@ -1,28 +1,41 @@
 package com.kodilla.good.patterns.flight;
 
-import java.util.Map;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public class FlightFinder {
 
-    public void findFlightFromAirport(String departureAirport, Map<Integer, Flight> availableFlightMap) {
-        System.out.println("\nAvailable flights form " + departureAirport + "\n");
-        availableFlightMap.entrySet().stream()
-                .filter(e -> e.getValue().getDepartureAirport().equals(departureAirport))
-                .forEach(System.out::println);
+    public List<Flight> departureAirportSearch(ArrayList<Flight> currentFlightsList, String selectedAirport) {
+        return currentFlightsList.stream().filter((n) -> n.getDepartureAirport().equals(selectedAirport)).collect(Collectors.toList());
     }
 
-    public void findFlightWithIntermediateAirport(String intermediateAirport, Map<Integer, Flight> availableFlightMap) {
-        System.out.println("\nAvailable flights through " + intermediateAirport);
-        availableFlightMap.entrySet().stream()
-                .filter(e -> e.getValue().getIntermediateAirport().equals(intermediateAirport))
-                .forEach(System.out::println);
+    public List<Flight> arrivalAirportSearch(ArrayList<Flight> currentFlightList, String selectedAirport) {
+        return currentFlightList.stream().filter((n) -> n.getArrivalAirport().equals(selectedAirport)).collect(Collectors.toList());
     }
 
-    public void findFlightToAirport(String arrivalAirport, Map<Integer, Flight> availableFlightMap) {
-        System.out.println("\nAvailable flights to " + arrivalAirport);
-        availableFlightMap.entrySet().stream()
-                .filter(e -> e.getValue().getArrivalAirport().equals(arrivalAirport))
-                .forEach(System.out::println);
+
+    public List<Flight> directFlightSearch(ArrayList<Flight> currentFlightList, String departureAirport, String arrivalAirport) {
+        List<Flight> flightsFromDestination = this.departureAirportSearch(currentFlightList, departureAirport);
+        return flightsFromDestination.stream().filter((n) -> n.getArrivalAirport().equals(arrivalAirport)).collect(Collectors.toList());
+    }
+
+    public HashMap<Flight, Flight> connectionFlightSearch(ArrayList<Flight> currentFlightList, String departureAirport, String arrivalAirport) {
```

Comment:
Zamodelowanie lotów łączonych przy pomocy `Map`-y nie jest poprawne. Jeśli w Twoim wyszukiwaniu znajdą się 2 różne wyniki, w których pierwszy lot jest ten sam, a różni się tylko drugi - to jeden z tych wyników nadpisze drugi.
Pomyśl o klasie reprezentującej pojedynczy wynik, np. `FlightSearchResult` i zwróć listę takich obiektów. Oczywiście tej samej klasy powinnać wtedy użyć do reprezentacji wyników pozostałych metod.